### PR TITLE
Addition of static methods for rpc

### PR DESCRIPTION
**Pull request by sarthak77**

### CODE REVIEW

**Review by rish691 — APPROVED** [review]

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/rpc/RpcSemanticConventionUtils.java [hunk 1]
```diff
@@ -328,4 +357,66 @@ public static String getGrpcErrorMsg(Event event) {
...
+  }
+
+  static Optional<String> getSanitizedAuthorityValue(String value) {
+    if (StringUtils.isBlank(value)) {
+      return Optional.empty();
+    }
+    // authority part of the uri assumes this format: <userinfo@host:port>
+    // parsing it as uri doesn't work, since this is not in url format
+    // allow for string of type <xyz> or <xyz:port> or <abc@xyz:port>
+    List<String> list = Splitter.on(COLON).splitToList(value);
```
Comment (jcchavezs) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/rpc/RpcSemanticConventionUtils.java:382]: Shall `Splitter.on(COLON)` be initialized every time?

Comment (jcchavezs) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/rpc/RpcSemanticConventionUtils.java:382]: name `list` is very generic and makes it hard to read in line 383. What is this list of? Can't we name it `uriPieces` or something like that?

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/rpc/RpcSemanticConventionUtils.java [hunk 2]
```diff
@@ -328,4 +357,66 @@ public static String getGrpcErrorMsg(Event event) {
...
+    if (list.size() <= 2 && !value.contains("/")) {
```
Comment (jcchavezs) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/rpc/RpcSemanticConventionUtils.java:383]: How about adding

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/rpc/RpcSemanticConventionUtils.java [hunk 3]
```diff
@@ -328,4 +357,66 @@ public static String getGrpcErrorMsg(Event event) {
...
+      String host = list.get(0);
+      Optional<String> port =
+          (list.size() == 2 && !StringUtils.isBlank(list.get(1)))
+              ? Optional.of(list.get(1))
+              : Optional.empty();
+      // reject string of type <:9000> where userinfo@host part is empty
+      if (StringUtils.isEmpty(host)) {
```
Comment (jcchavezs) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/rpc/RpcSemanticConventionUtils.java:390]: Shall this happen right after we declare host?

**Pull request merged**